libtenzir/builtins/operators/http.cpp

```diff
@@ -366,12 +427,44 @@ auto find_plugin_for_mime(std::string_view mime, location op,
 }
 
 auto make_pipeline(const std::optional<located<pipeline>>& pipe,
-                   const http::response& r, location oploc,
+                   const caf::uri& uri, const http::response& r, location oploc,
                    diagnostic_handler& dh, bool is_warning)
   -> failure_or<located<pipeline>> {
   if (pipe) {
     return *pipe;
   }
+  auto parsed = boost::urls::parse_uri_reference(uri.str());
+  if (not parsed) {
+    diagnostic::error("invalid URI `{}`", uri.str()).primary(oploc).emit(dh);
+    return failure::promise();
+  }
```

Comment:
How does this interact with the `http` operator, which probably wants warnings here?

Reply:
Any there other errors that `http` can emit while the pipeline is running? This should probably be a warning like every other case where we can continue afterwards.

Reply:
Failing to make the request, for example due to network issues is also an error.

Reply:
Or what do you think? Are there good reasons for `http` to emit errors in such cases, and do they outweigh the disadvantages of doing so?

Reply:
Anything that can be a warning should be a warning, that's my general sentiment. So unless we _must_ stop execution, it should be a warning and not an error.